### PR TITLE
Multi-arch docker image build via GitHub Actions

### LINKED ISSUE #200: ARM builds broken
rpi4, docker, installed from nightly
first run i dont have access to all options in UI
https://youtu.be/NEkSKtG9HPM

### PR DESCRIPTION
**Pull request by diveflo**
Hi,

this PR would enable automated GitHub Actions-based docker builds for multiple architectures via [docker buildx](https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images).
The resulting docker images are multi-platform themselves, i.e., you don't need separate tags for different CPU architectures. Using `latest` just works out of the box for all supported CPU architectures.

Thankfully, your `Dockerfile` actually already works for these builds, as `alpine` is now also a multi-arch image, and the separate tag for `arm` isn't required anymore.

With the current configuration, this would create images that support `linux/amd64`, `linux/arm`, and `linux/arm64/v8`, but you could add additional platforms.

The build would be moved from Dockerhub's own builds to GitHub Actions, as Dockerhub doesn't support this yet.
Hence, you would need to add authentication to this Repo via Secrets, so that the Github Actions build can authenticate against Dockerhub to push the resulting images.
Basically you would need to add these two Secrets, where the first one would be your Dockerhub user name and the second one an access token. See this example from my fork of your repo:

2020-11-21 20_19_53-Window

The PR is using the official and currently recommended Docker actions, see [here](https://github.com/marketplace/actions/build-and-push-docker-images).

One caveat: I'm not sure how fast your current builds on Dockerhub are, but for the three platforms above, the GitHub Actions build takes around ~1h.

Opening this as a draft as I still want to test the arm64 image locally.

You can see how the build would look in my fork:
https://github.com/diveflo/YoutubeDL-Material/runs/1436354825?check_suite_focus=true

and how the resulting multi-arch image would look:
https://hub.docker.com/repository/docker/floriang89/youtubedl-material/tags?page=1

Let me know what you think:)

Edit:
This should fix #200

### CODE REVIEW

**Comment by diveflo** [comment]
The `docker / build-and-push (pull_request)` job failed as the required GitHub secrets are missing obviously 😉

**Comment by Tzahi12345** [comment]
Awesome PR! I'll gladly merge, but I have one request: can you make it so that the autobuilds only apply to the nightly tag? Latest is currently the latest release. Right now it's an issue because latest is a few months old, but once 4.2 drops it won't be so outdated.

But yeah this improvement is great, the arm nightly builds are hella outdated since the autobuild stopped working a few months ago as well

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: build & push images
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm,linux/arm64/v8
+          push: true
+          tags: tzahi12345/youtubedl-material:latest
+
```
Comment (SuperSandro2000) [inline .github/workflows/docker.yml:32]:
```suggestion

```

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: build & push images
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm,linux/arm64/v8
+          push: true
+          tags: tzahi12345/youtubedl-material:latest
```
Comment (SuperSandro2000) [inline .github/workflows/docker.yml:31]:
```suggestion
          tags: tzahi12345/youtubedl-material:nightly
```

**Comment by SuperSandro2000** [comment]
Can this also push the release tags?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -28,5 +28,5 @@ jobs:
           file: ./Dockerfile
           platforms: linux/amd64,linux/arm,linux/arm64/v8
           push: true
-          tags: tzahi12345/youtubedl-material:latest
-  
\ No newline at end of file
+          tags: tzahi12345/youtubedl-material:nightly
+
```
Comment (SuperSandro2000) [inline .github/workflows/docker.yml:32]:
```suggestion
```

I think there are still two final new lines.

Comment (Tzahi12345) [reply .github/workflows/docker.yml]: Yeah I think GitHub warns if it's EOF without a new line

**Comment by Tzahi12345** [comment]
Just did a final smoke test on your image, it looks good! I'll let you test it on ARM, but once that checks out I'll happily merge. I'm not able to test the actual build process even with my secrets because they don't apply to PR's from other user's repos (complain to GitHub for that one lol)

@SuperSandro2000 this [looks possible](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#release), but I haven't played around with much of the GitHub workflow environment so don't quote me on that.

**Comment by diveflo** [comment]
Wow - didn't expect such quick replies:)
I tested it on both arm64 and arm/v8 and both work!

So let's get this merged and hopefully then your secrets apply to the pipeline 👍

There's also ways of using this for releases. How's your usual workflow for this? Do you want releases for pushed tags with a specific format, e.g., every pushed tag with "v*" get's released?

**Comment by Tzahi12345** [comment]
>Oh yeah makes sense that GitHub isn't applying your secrets to pull requests. Because I could just add echo ${{secrets.DOCKHUB_TOKEN}} to the workflow file and get your token 😅

Haha yeah I didn't really consider that. I'll merge and get back to you on the workflow stuff regarding the release builds once I review your other PR -- great work on this one @diveflo!

**Comment by diveflo** [comment]
great:)
If the images work, you could disable the automated builds on dockerhub, so this mean red x goes away :D
image

**Pull request merged**